yt_dlp/extractor/crunchyroll.py

```diff
@@ -359,31 +383,15 @@ def entries():
         else:
             raise ExtractorError(f'Unknown object type {object_type}')
 
-        # There might be multiple audio languages for one object (`<object>_metadata.versions`),
-        # so we need to get the id from `streams_link` instead or we dont know which language to choose
-        streams_link = response.get('streams_link')
-        if not streams_link and traverse_obj(response, (f'{object_type}_metadata', 'is_premium_only')):
+        if not self._IS_PREMIUM and traverse_obj(response, (f'{object_type}_metadata', 'is_premium_only')):
             message = f'This {object_type} is for premium members only'
             if self.is_logged_in:
                 raise ExtractorError(message, expected=True)
             self.raise_login_required(message)
 
-        # We need go from unsigned to signed api to avoid getting soft banned
```

Comment:
have you / has anyone tested if many logged-in requests to the switch endpoint results in a soft ban? (like unsigned cms endpoint requests would do before)

Reply:
I haven't explicitly tried, but for testing I requested the endpoint like 20 - 30 times in a short time period and didn't get a ban. I'll remove the comment